test: completes code coverage in `blas/base/dnrm2`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds test cases for `blas/base/dnrm2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-  none 

## Questions

> Any questions for reviewers of this pull request?

Is the tape name that I have given proper?

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
test: completes code coverage in blas/base/dnrm2

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: update test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
fix: update test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.dnrm2.js:
```diff
@@ -45,14 +76,39 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
 	x = new Float64Array( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
 
-	z = dnrm2( x.length, x, 1 );
+	z = dnrm2( x.length, x, 1, 0 );
```
The main export does not have an offset parameter. Applies here and below.

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.dnrm2.native.js:
```diff
@@ -54,14 +85,39 @@ tape( 'the function calculates the L2-norm of a vector', opts, function test( t
 	x = new Float64Array( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
 
-	z = dnrm2( x.length, x, 1 );
+	z = dnrm2( x.length, x, 1, 0 );
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.dnrm2.js:
```diff
@@ -45,14 +76,39 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
+	z = dnrm2( x.length, x, 1, 0 );
 	t.strictEqual( z, 4.0, 'returns expected value' );
 
+	x = new Float64Array([ 1e150, 1e150, 1e150, 1e150 ]);
```
```suggestion
	x = new Float64Array( [ 1e150, 1e150, 1e150, 1e150 ] );
```

When on a single line, include spaces. Applies here and elsewhere.

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.ndarray.js:
```diff
@@ -53,6 +84,31 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
 	z = dnrm2( x.length, x, 1, 0 );
 	t.strictEqual( z, 4.0, 'returns expected value' );
 
+	x = new Float64Array([ 1e150, 1e150, 1e150, 1e150 ]);
```
Spacing.

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.dnrm2.js:
```diff
@@ -45,14 +76,39 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
+	z = dnrm2( x.length, x, 1, 0 );
+	t.strictEqual( z, 2.0e+150, 'returns expected value' );
+
+	x = new Float64Array([ 1e-155, 1e-155, 1e-155, 1e-155 ]);
```
```suggestion
	x = new Float64Array( [ 1.0e-155, 1.0e-155, 1.0e-155, 1.0e-155 ] );
```

Use decimals for representing double-precision decimal values. Applies here and elsewhere.

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.ndarray.js:
```diff
@@ -53,6 +84,31 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
+	x = new Float64Array([ 1e-155, 1e50, 1e-155, 1e50 ]);
+
+	z = dnrm2( x.length, x, 1, 0 );
+	isApprox( t, z, 1.41421e50, 2.0 );
```
Why are you using incomplete decimal values? Here you go only to 5 decimals. If you extend to full precision, can you decrease your test tolerance? Will the results equal the exact expected result?

kgryte replied:
@ShabiShett07 This question still remains.

kgryte reviewed (changes requested):
Left an initial round of comments.

ShabiShett07 committed:
fix: add decimal and remove offset

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: add decimal and remove offset

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: add decimals

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: add decimals

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.dnrm2.js:
```diff
@@ -53,6 +84,31 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
 	z = dnrm2( x.length, x, 1 );
 	t.strictEqual( z, 4.0, 'returns expected value' );
 
+	x = new Float64Array([ 1.0e150, 1.0e150, 1.0e150, 1.0e150 ]);
```
```suggestion
	x = new Float64Array( [ 1.0e150, 1.0e150, 1.0e150, 1.0e150 ] );
```

@ShabiShett07 You are still missing spacing inside parentheses here and elsewhere throughout this PR.

ShabiShett07 committed:
fix:  indentations

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: add indentations

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: increase test precision

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I have extended the precision upto 16 decimals and also reduced the tolerance to 1 epsilon as per your suggestion

kgryte commented on lib/node_modules/@stdlib/blas/base/dnrm2/test/test.ndarray.native.js:
```diff
@@ -62,6 +93,31 @@ tape( 'the function calculates the L2-norm of a vector', opts, function test( t
+	x = new Float64Array( [ 1.0e150, 1.0e50, 1.0e150, 1.0e50 ] );
+
+	z = dnrm2( x.length, x, 1, 0 );
+	isApprox( t, z, 1.41421e150, 2.0 );
```
@ShabiShett07 I believe you still need to adjust the precision in this file, as well, correct?

ShabiShett07 committed:
chore: update test case

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I think now it is ready to merge
